src/auth/storacha.ts

```diff
@@ -0,0 +1,151 @@
+import * as Client from "@web3-storage/w3up-client";
+import { Config, readConfig, writeConfig } from "../utils/config";
+import inquirer from "inquirer";
+import ora from "ora";
+import chalk from "chalk";
+import { OwnedSpace, Space } from "@web3-storage/w3up-client/dist/src/space";
+
+export class StorachaAuth {
+  private config: Config;
+
+  constructor() {
+    this.config = readConfig();
+  }
+
+  async login(): Promise<Client.Client> {
+    const { email } = await inquirer.prompt([
+      {
+        type: "input",
+        name: "email",
+        message: "Enter your email:",
+        validate: (input) => {
+          if (!input.includes("@")) {
+            return "Please enter a valid email address";
+          }
+          return true;
+        },
+      },
+    ]);
+
+    const spinner = ora("Logging in to Storacha...").start();
+    spinner.color = "red";
+
+    try {
+      const client = await Client.create();
+      await client.login(email);
```

Comment:
this may work, but you should probably use the `Account.login` method like we do in w3cli:

https://github.com/storacha/w3cli/blob/3a737891c4ad988e091eca9d0fc9bc96a4f83a6f/account.js#L77

Reply:
okay. thank you! I'll check this too